### ISSUE TITLE
Facebook API October 24th changes 

### ISSUE THREAD

**Issue by olivM** [oscarotero/Embed#392]
Hi,

Is there a branch planned to test the upcoming change to Facebook/Intagram API endpoint ?

https://developers.facebook.com/docs/graph-api/changelog/version8.0#instagram
https://developers.facebook.com/docs/graph-api/changelog/version8.0#social-plugins

thanks

**Comment by oscarotero** [comment]
Thanks for notify this.
Seems like now facebook require an access token: https://developers.facebook.com/docs/plugins/oembed

I don't have much free time now so, if anyone want to work on this, PR is appreciated. If not, I'll work on it in next weeks.

**Comment by oscarotero** [comment]
I've added a couple of new settings to configure the access tokens for Facebook and Instagram (https://github.com/oscarotero/Embed#settings)

Could you please test if it works fine? I don't have access tokens for that and I don't want to create an app for that.

Thanks!

**Comment by reflexxion** [comment]
> I've added a couple of new settings to configure the access tokens for Facebook and Instagram (https://github.com/oscarotero/Embed#settings)
> 
> Could you please test if it works fine? I don't have access tokens for that and I don't want to create an app for that.
> 
> Thanks!

If I'm testing it right, it does not seem to work.
When the Extractor is present, the detectEndpoint method was called before I was able to set the settings.
A PR is incoming.

**Comment by Rapid0s** [comment]
Endpoints have to be updated also for tokens have any meaning

**Comment by oscarotero** [comment]
Released this change in https://github.com/oscarotero/Embed/releases/tag/v4.3.0
Thanks @reflexxion

**Comment by BSimon47** [comment]
May I inquire if this discussion covers the Drupal Module: https://github.com/oscarotero/Embed/issues/392 ???

The resolution to this is of the greatest importance to finalizing the release of a website that has been a full year in the making and this policy change by Facebook has stalled the release.

Thank you Oscar, for both your Oembed program and resolving this!

**Comment by oscarotero** [comment]
We have updated Embed v3 & v4.
According to this https://github.com/oscarotero/Embed/issues/362 Drupal still using v2, a very old and unmaintained version of this library. I encourage to Drupal plugin creators to update the plugin to use v3 (or even better, v4).

**Comment by BSimon47** [comment]
@oscarotero - I just updated my Drupal distribution and can confirm that it updated embed/embed to version 3.4.10 however, the Facebook issue still exists. The URL entered renders as a URL only vs the intended card.

Your thoughts?

**Comment by oscarotero** [comment]
@BSimon47 Facebook and Instagram now require an api key to use its oembed api (here the docs: https://github.com/oscarotero/Embed/tree/v3.x#example-with-all-options)
I don't know how this is integrated with Drupal but there should be a way to configure an api key for this.

**Comment by BSimon47** [comment]
Hey Oscar,

I have a question for you.
If I embed the following URL ( https://variety.com/2020/film/news/directors-guild-warner-bros-hbo-max-deal-1234849614/ ), I get an error in my Drupal Embed which has an updated Embed Library to drupal/url_embed: ^3.4.10


Even your own Test Site shows a similar error



If I updated the library to the v4.0 branch would that resolve this issue?

And if so, where do I find that code for the library.

Part of the reason I ask (aside from my own purposes) is that I have been in communication with the Drupal Developers for the URL_Embed module (Who are seeking maintainers – of which I am not qualified) and passing along your notes with specific attention to the Facebook/Instagram issue.

Thanks in advance

 



Email:  <mailto:bas@crew-list.net> bas@crew-list.net

**Comment by oscarotero** [comment]
You can test any url in the version 4.x here: https://oscarotero.com/embed/demo/index.php?url=https%3A%2F%2Fvariety.com%2F2020%2Ffilm%2Fnews%2Fdirectors-guild-warner-bros-hbo-max-deal-1234849614%2F&settings=

Version 3.x is here: https://oscarotero.com/embed3/demo/index.php?url=https%3A%2F%2Fvariety.com%2F2020%2Ffilm%2Fnews%2Fdirectors-guild-warner-bros-hbo-max-deal-1234849614%2F

The error in the embed code is from the website, not related with this library (that only returns the code provided by the website).

**Comment by BSimon47** [comment]
Oscar, I understand that and appreciate your reply.  I do not see any difference in this case between your versions as clearly the callback from Variety denies the card and instead issues the denial code which makes the site looking to share their content look bad! 

What I am trying to express is a concern that I believe you might share. What are we to do when so many websites such as Variety.com and Deadline.com etc. return these denials. I am trying to provide my users who fit directly into the same demographic as Variety.com and many others who are now adopting these policies of denial, a method of adding substance to back up their posts in much the same way that Facebook and Twitter do.

Certainly in my mind it defeats the purpose of promoting their content with a simple teaser card that links back to their original articles.

All of that said, what is ironic is that (once the Facebook issues are resolved) I could simply share their content on Facebook in a post and then share that card in my site and it would work just as if I had shared their URL.

Well, hopefully the Drupal url_embed folks will make the necessary changes to reinstate the Facebook/Instagram API’s to process the Developer Key and Token sooner than later. I have contacted two of their main contributors directly in the hopes that something will happen.

This tool is far too important to go to waste in this way! It serves as a win win in content sharing for all platforms in the chain!

Regards,

 



Ph: 818.414.1795

Email:  <mailto:bas@crew-list.net> bas@crew-list.net
